### PR TITLE
Update contributing.md for namespace

### PR DESCRIPTION
**Pull request by chris-ashe**
This pull request introduces several updates to the documentation and templates to ensure adherence to the PROCESS style guide and improve overall readability and consistency. The most important changes include adding references to the PROCESS style guide in various files, expanding the style guide documentation, and updating naming conventions in the codebase.

### Documentation and Templates Updates:

* **Pull Request Template**: Added a checkbox to confirm that changes follow the PROCESS style guide. (`.github/pull_request_template.md`)
* **Contributing Guide**: Updated to include a reference to the PROCESS style guide for naming variables and functions. (`CONTRIBUTING.md`)

### Style Guide Enhancements:

* **Style Guide Introduction**: Added an introduction to the PROCESS style guide, including adherence to PEP8 for Python and namespace style. (`documentation/proc-pages/development/standards.md`)
* **Naming Conventions**: Expanded section on naming conventions for functions, switches, constants, and variables, including guidelines for variables representing fractions. (`documentation/proc-pages/development/standards.md`)
* **Units and Types**: Updated examples and conventions for units and type hints in Python, as well as added new sections for docstrings and comments. (`documentation/proc-pages/development/standards.md`) 

### Navigation Update:

* **MkDocs Configuration**: Changed the navigation label from "Reference" to "Models" in the documentation site configuration. (`mkdocs.yml`)

## Checklist

I confirm that I have completed the following checks:

- [ ] I have justified any large differences in the regression tests caused by this pull request in the comments.
- [ ] I have added new tests where appropriate for the changes I have made.
- [ ] If I have had to change any existing unit or integration tests, I have justified this change in the pull request comments.
- [ ] If I have made documentation changes, I have checked they render correctly.
- [ ] I have added documentation for my change, if appropriate.

### CODE REVIEW

File: documentation/proc-pages/development/standards.md [hunk 1]
```diff
@@ -1,11 +1,17 @@
 # Style Guide
 
+`PROCESS` follows the Python [PEP8](https://peps.python.org/pep-0008/) style guide for its layout and namespace style. This should not conflict with Fortran90 in terms of names but may with the line lengths.
```
Comment (jonmaddock) [inline documentation/proc-pages/development/standards.md:3]: Could you state what the line length limit is, and maybe provide an example?

File: documentation/proc-pages/development/standards.md [hunk 2]
```diff
@@ -20,16 +26,46 @@ real(dp) :: b
...
+    *"Readability counts."*
+
+### Functions
+
+Use a lowercase word or words. Separate words by underscores(`_`) to improve readability.
+
+### Switches
+
+Switches should start with the `i_` prefix in their name, be of integer type and should be indexed from 0.
+Switches should by their nature not override other switches.
```
Comment (jonmaddock) [inline documentation/proc-pages/development/standards.md:48]: Could you clarify this? There are current cases (and legitimate future ones) where defining a switch as one value means that others are ignored or are invalid. Perhaps something could be stated about these scenarios being caught, so switches aren't tacitly ignored in this case, but instead should throw exceptions?

File: documentation/proc-pages/development/standards.md [hunk 3]
```diff
@@ -20,16 +26,46 @@ real(dp) :: b
...
+If the constants doesn't exist then add it with a source link and uncertainty value.
+
+### Variables
+
+Use a lowercase single letter, word, or words. Separate words with underscores to improve readability.
+If converting between units it may be required to have some capital letters at the end of the variable name to differentiate between different orders of magnitude, `m` and `M` etc.
+
+#### Variables representing fractions
+
+If a variable is intended to demonstrate a fraction of a value or distribution etc. Then it should start with the `f_` prefix.
```
Comment (jonmaddock) [inline documentation/proc-pages/development/standards.md:64]: What about f-values, which also follow this pattern (at the moment)?

File: documentation/proc-pages/development/standards.md [hunk 4]
```diff
@@ -94,32 +132,102 @@ ii
...
+
+    Notes:
+        Detailed description of the function. This can include information about the algorithm,
+        any important notes, and other relevant details.
+
+    References:
+        - Reference 1 
+        - Reference 2
+         (Use a verbose reference style that has a clickable DOI or HTML link.)
+    """
```
Comment (jonmaddock) [inline documentation/proc-pages/development/standards.md:190]: I believe this is a modified version of the numpy (or is it Google?) docstring format; currently we use the sphinx/reStructuredText one. I'd be keen to use one format, and then convert using a tool (pretty sure I've found ones before) if we want a different one. I don't particularly like the sphinx one, but it's widely used. Perhaps make it explicit which format we follow here, and reference the autodocstring (or equivalent) extensions in VSCode.

**Review by jonmaddock — CHANGES_REQUESTED** [review]
Thanks for doing this Chris, I appreciate it. Some minor clarifications please.

File: documentation/proc-pages/development/standards.md (hunk 1 above)
Comment (chris-ashe) [reply documentation/proc-pages/development/standards.md:3]: This is what I wanted to know about for the Fortran and how we may be doing it differently.
For PEP 8 it is:

"Limit all lines to a maximum of 79 characters."

"For flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to 72 characters."

File: documentation/proc-pages/development/standards.md (hunk 3 above)
Comment (chris-ashe) [reply documentation/proc-pages/development/standards.md:64]: Good point, I think for this case the majority of f-values have the `f` prefix without the underscore. There is currently quite a mish mash of fractional variables and f-values. Though hopefully soon with the number of f-values being reduced and variables being renamed to have the proper `f_` prefix, this should be more clear

File: documentation/proc-pages/development/standards.md (hunk 4 above)
Comment (chris-ashe) [reply documentation/proc-pages/development/standards.md:190]: It is a variant of the Google docstring style, though with some additions as we cite papers etc. I will add this note and also a possible extension

File: documentation/proc-pages/development/standards.md (hunk 1 above)
Comment (jonmaddock) [reply documentation/proc-pages/development/standards.md:3]: I believe there is a line length limit in Fortran, but it's bigger than the PEP8 recommendations. I'd just include what you've commented above: this will probably be enforced by a formatter at some point anyway.

File: documentation/proc-pages/development/standards.md [hunk 5]
```diff
@@ -163,6 +164,20 @@ It is greatly encouraged and recommended to include type hints for all inputs an
 
 ## Docstrings
 
+The docstring style is that of the [Google type](https://google.github.io/styleguide/pyguide.html#s3.8.1-comments-in-doc-strings). Though there are some additions for `Notes` and `References` in order to give mathematical reasoning and sources to some functions.
```
Comment (jonmaddock) [inline documentation/proc-pages/development/standards.md:167]: I really meant not to start including Google-style docstrings when the majority of the codebase already uses sphinx-style. I think ff we want to switch over to another one, this should be done in a separate PR, hopefully using a tool to convert all existing docstrings.

**Review by jonmaddock — CHANGES_REQUESTED** [review]
Brilliant Chris, I think you've addressed everything apart from the conflicting docstring formats. Much clearer!

File: documentation/proc-pages/development/standards.md (hunk 5 above)
Comment (chris-ashe) [reply documentation/proc-pages/development/standards.md:167]: I will amend

**Review by jonmaddock — APPROVED** [review]
